### ISSUE TITLE
Bug: 'Please check your Primary Site's Jetpack connection' error sometimes directs to incorrect wp-admin, blocking users from fixing

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#79324]
This issue was reported here: https://github.com/Automattic/jpop-issues/issues/8387
Then merged over into this feature request: https://github.com/Automattic/wp-calypso/issues/56550
I think we can still do a better job tracking and triaging this as it's own report, so I'm recreating here. 

---

### Quick summary

Users are stuck on wordpress.com/stats/day/ with the notice `Please check your Primary Site's Jetpack connection` when either no primary site selected or primary site legitimately disconnected from Jetpack (i.e. old self hosted site that's no longer in use).

![Markup 2023-06-25 at 10 03 58](https://github.com/Automattic/wp-calypso/assets/27249804/ce661e70-faaa-4a43-85fc-a67d46500304)

The `wp-admin` link shown on the error tries to link to a different site's dashboard.

From @Nic-Sevic 

> wpcom users are not able to resolve this themselves and shouldn't be seeing this message in the first place. The issue appears to be that the correct assignment of the primary site isn't happening. Users could potentially fix this on their own by changing the primary site in their account but there's nothing to indicate this to them.
- https://github.com/Automattic/wp-calypso/issues/56550#issuecomment-1630202134

And

> It looks like the "Please check your Primary Site's Jetpack connection" error should show up only when user has one site and that one isn't reachable: fbhepr%2Skers%2Spnylcfb%2Spyvrag%2Szl%2Qfvgrf%2Spbagebyyre.wf%3Se%3Q0639q560%23546-og
> In these cases, users had other sites connected so I'm not sure how they got here. That may be the reason this is so hard to reproduce.
- https://github.com/Automattic/jpop-issues/issues/8387#issuecomment-1624148528

### Steps to reproduce

Steps to reproduce have not been nailed down yet. 

### A clear and concise description of what you expected to happen.

Primary site should default to one that's live if the previous value no longer exists
Even if primary site not available (not connected to JP), when you're trying to go to another site it shouldn't matter

### What actually happened

Regardless of site user is trying to go to, they get the `Please check your Primary Site's Jetpack connection` notice and then have no indication of what's wrong.

### Impact

Some (< 50%)

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Atomic

**Comment by cuemarie** [comment]
Recent interactions noted on https://github.com/Automattic/wp-calypso/issues/56550 :

1. 6460967-zd-woothemes - Troubleshooting Thread: p1687689160318619-slack-C03TY6J1A
2. 6463319-zd-woothemes - Troubleshooting Thread: p1687678991583859-slack-C03TY6J1A
3. 6521896-zd-woothemes - Troubleshooting Thread: p1689029333516649-slack-C03TY6J1A

**Comment by cuemarie** [comment]
In 2 of the above slack threads (1 & 3), HEs noted that JP Debug reported the site's JP connection was healthy, and performing the typical troubleshooting (plugin/theme conflict test, swap primary domain, dis/reconnect Jetpack) did not resolve. 

In all 3, it was found that the `wp-admin` link in the error directed to the wp-admin dashboard of the wrong site, and disconnecting Jetpack from that that wrong site resolved the issue.

**Comment by cuemarie** [comment]
📌 ACTIONS
- Needs reply

📌 Message
- @jeherve Can you identify any way we might be able to replicate this?  Anecdotally, I've certainly seen this error screen that user's can't get past plenty during support, but it's been hard to repro.  Some key details:

- JP Debug reports Healthy for the site in question
- The wp-admin link is directing to a wrong/old/disconnected site (which is still associated with JP)
- Disconnecting the wrong site from JP seems to help resolve

Perhaps there's a way we can try to intentionally create a Jetpack IDC, with the domain that's used for both sites set as the user's primary site?

**Comment by jeherve** [comment]
> Can you identify any way we might be able to replicate this? 

I'll let @bindlegirl chime in since she worked on the original report.

**Comment by cuemarie** [comment]
This came up again today here: p1689365540110789-slack-C03TY6J1A

Having the user set their correct site as the primary site under https://wordpress.com/me/account resolved the error in the dashboard. 

- The dashboard still shows there are 2 sites, but with 1 hidden, bc a JP connection remains on their old domain. 
- They cancelled the old domain in January and registered a new one the same day.
- The Primary Site's Jetpack Connection warning did not start then, however.  We asked the user when it began and they shared:

> "I don't recall to be honest. I did notice the change happened around the time when I downloaded jetpack on my phone so I can edit my site on there"

**Comment by Nic-Sevic** [comment]
another here: 6543164-zen

**Comment by bindlegirl** [comment]
> I'll let @bindlegirl chime in since she worked on the original report.

I tried different things to reproduce this issue but no luck so far. I have tried causing IDC, but all I got was a "This site cannot be accessed." notice with an option to fix things.

I tried other ways of breaking & deleting a primary site, but nothing caused this error. In every scenario, I get that same notification, but the existing stats get loaded.

**Comment by Nic-Sevic** [comment]
another here: 6596433-zd-a8c

**Comment by cuemarie** [comment]
While working with KitKat to test and triage this old issue: https://github.com/Automattic/wp-calypso/issues/24539, I reached the `This site cannot be accessed` stage of the issue by taking the following steps:

- New Self-Hosted Site (Pressable), with primary Jetpack connection to a new WPcom user: `testcuemarie`
- `testcuemarie` set the site as the primary site under `/me/account`
- `testcuemarie` created a second, free, WPcom site
- In Pressable, I deleted the self-hosted site. 

When I just disconnect Jetpack via JP Debugger, the `/me/account` primary site switches to the other site, but when I delete the site via the host, the primary site doesn't change - this seems like it could be the first step to the problem.

From here, I tried archiving & deleting the site via NA, but both those actions removed the SH site from my `testcuemarie`'s dashboard and setting the other site as the "primary site" in `/me/account`, effectively resolving the 
issue. 

**Next**

- Just going to leave this (now undeleted from NA) site as it is - perhaps in time we'll see the behavior change and trigger the error above.

**Comment by poeticabdul** [comment]
Reported here 6794497-zd-a8c

**Comment by cuemarie** [comment]
📌 Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO

I have replicated this problem!  The account I set up on Aug 8 (https://github.com/Automattic/wp-calypso/issues/79324#issuecomment-1670134262 above) now shows the `Please check your Primary Site's Jetpack connection` error when I log in!

![Markup on 2023-09-28 at 14:12:37](https://github.com/Automattic/wp-calypso/assets/27249804/ebc8539a-ebbe-4c29-a81b-ab8a052f27bc)

The `wp-admin` link directs to the self-hosted site that I'd left disconnected from Jetpack and set as the Primary site under Account settings.

In that site's BRC, this is the last action, on `2023-09-13`:

2w |   | add sticker | 0 | is_disconnected \| Force Sticker: No contact for 30 days | no-ip
-- | -- | -- | -- | -- | --

---

TLDR - On 2023-08-08 I undeleted the account's primary site, but left it with a broken JP connection. On `2023-09-13`, the No contact for 30 days sticker was added, and now I am encountering the error above. 

The `/me/account` page shows a blank selection under **Primary site**, and the primary site in question is hidden from that list.

![Markup on 2023-09-28 at 14:19:17](https://github.com/Automattic/wp-calypso/assets/27249804/2ca45dad-5e76-4d5f-917d-daf12b68bc46)

![Markup on 2023-09-28 at 14:19:41](https://github.com/Automattic/wp-calypso/assets/27249804/6d78803a-9a1b-4686-b9d0-005e3e5b61bf)

---

📌 ACTIONS
- Lowering priority to `Normal` for now

📌 Message to Author/Devs
@bindlegirl @Nic-Sevic I won't change anything on this account set up for now - let me know if I can provide any other details or access to the site so you can mess around with it! 

Account ID: `238512620`
Disconnected (primary) site ID: `222159544`

**Comment by kuropixel** [comment]
Encountered on 7109554-zen

**Comment by Nic-Sevic** [comment]
happened again on a simple site (was trying to access a self hosted site which no longer exists) - p9F6qB-dT4-p2

**Comment by jorpdesigns** [comment]
Encountered in 7635994-zen

**Comment by supernovia** [comment]
I encountered a possibly similar case here:
7682802-zd-a8c

In this one I experienced the error intermittently :/

**Comment by galakhyati** [comment]
Encountered a similar case here: p1706692895589939-slack-C03TY6J1A

**Comment by retnonindya** [comment]
@bindlegirl Hi! Do we have any updates on this? It seems like there are several reports surfacing again lately, and I pinged you here because I noticed you have been checking on this issue. Let me know if this better be brought to a different channel 🙇  Thanks!

**Comment by Nic-Sevic** [comment]
another case: 7704145-zd-a8c

**Comment by Nic-Sevic** [comment]
another instance: p1709117508244669-slack-C03TY6J1A

**Comment by davemart-in** [comment]
Seems like there were a rash of reports, but no cases reported in some time. I'm going to close this out. But happy to re-open it if this is still an issue.
